src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc

```diff
@@ -400,7 +401,21 @@ void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
       errors_.AddError(service_config.status().message());
       service_config_json_ = service_config.status();
     } else {
-      service_config_json_ = absl::StrCat("grpc_config=", *service_config);
+      constexpr char kServiceConfigAttributePrefix[] = "grpc_config=";
+      auto result = std::find_if(service_config->begin(), service_config->end(),
+                                 [&](absl::string_view s) {
+                                   return absl::StartsWith(
+                                       s, kServiceConfigAttributePrefix);
+                                 });
+      if (result != service_config->end()) {
```

Comment:
This appears to differ from previous logic where we simply concatenate the grpc_config prefix to all returned values. Instead now it searches for a result with the given prefix. Is there a specific reason ?

Reply:
yeah, I think the previous logic is not correct (this code is still under development and not used until EventEngine::DNSResolver is ready) since gRPC's service config data starts with `grpc_config=` followed by the JSON-formatted data. See: https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md#encoding-in-dns-txt-records.